apps/documents/tests/test_views.py

```diff
@@ -0,0 +1,72 @@
+from unittest import mock
+
+import pytest
+from django.urls import reverse
+
+from apps.documents.models import CollectionFile, FileStatus
+from apps.utils.factories.documents import CollectionFactory
+from apps.utils.factories.service_provider_factories import LlmProviderFactory
+from apps.utils.factories.team import TeamWithUsersFactory
+
+
+@pytest.mark.django_db()
+class TestEditCollection:
+    @pytest.fixture()
+    def collection(self):
+        team = TeamWithUsersFactory()
+        return CollectionFactory(name="Tester", team=team, is_index=True, llm_provider=LlmProviderFactory(team=team))
+
+    @mock.patch("apps.documents.views.OpenAIVectorStoreManager")
+    @mock.patch("apps.documents.tasks.migrate_vector_stores.delay")
+    def test_update_collection_with_llm_provider_change(self, migrate_mock, manager_mock, collection, client):
+        new_llm_provider = LlmProviderFactory(team=collection.team)
+        new_vector_store_id = "new-store-123"
+
+        manager_instance = mock.Mock()
+        manager_instance.create_vector_store.return_value = new_vector_store_id
+        manager_mock.from_llm_provider.return_value = manager_instance
+
+        client.force_login(collection.team.members.first())
+        url = reverse("documents:collection_edit", args=[collection.team.slug, collection.id])
+
+        response = client.post(
+            url,
+            {
+                "name": collection.name,
+                "is_index": True,
+                "llm_provider": new_llm_provider.id,
+            },
+        )
+
+        assert response.status_code == 302
+        collection.refresh_from_db()
+        assert collection.llm_provider == new_llm_provider
+        assert collection.openai_vector_store_id == new_vector_store_id
+
+        # Verify that files are marked for reprocessing
+        CollectionFile.objects.filter(collection=collection).update(status=FileStatus.PENDING)
+
+        # Verify migration task was called
+        migrate_mock.assert_called_once_with(
+            collection_id=collection.id,
+            from_vector_store_id=mock.ANY,
+            from_llm_provider_id=mock.ANY,
+        )
+
+    def test_update_collection_without_llm_provider_change(self, collection, client):
+        client.force_login(collection.team.members.first())
+        url = reverse("documents:collection_edit", args=[collection.team.slug, collection.id])
+        new_name = "Updated Collection Name"
+
+        response = client.post(
+            url,
+            {
+                "name": new_name,
+                "is_index": True,
+                "llm_provider": collection.llm_provider.id,
+            },
+        )
+
+        assert response.status_code == 302
+        collection.refresh_from_db()
+        assert collection.name == new_name
```

Comment:
could check that `migrate_vector_stores` is not called here

Reply:
https://github.com/dimagi/open-chat-studio/pull/1571/commits/46f8b7d906e6aa2e468196c59501d0c7bf9bf5b4